libs/api_common/windows_platform_common.cpp

```diff
@@ -136,10 +456,293 @@ get_attach_type_windows(const std::string& section)
-    auto it = windows_section_names.find(*attach_type);
-    if (it != windows_section_names.end())
-        return it->second;
+    for (const auto& t : _windows_section_definitions) {
+        if (IsEqualGUID(*t.get()->attach_type, *attach_type)) {
+            return t.get()->section_prefix;
+        }
+    }
+
+    return nullptr;
+}
+
+static ebpf_result_t
+_update_global_helpers_for_program_information(
+    _In_reads_(global_helper_count) const ebpf_helper_function_prototype_t* global_helpers,
+    uint32_t global_helper_count)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_helper_function_prototype_t* new_helpers = nullptr;
+    size_t total_helper_count = 0;
+    size_t total_helper_size = 0;
+
+    if (global_helper_count == 0) {
+        return result;
+    }
+
+    // Iterate over all the program information and append the global
+    // helper functions to each of the program information.
+    for (auto& iterator : _windows_program_information) {
+        ebpf_program_info_t* program_info = iterator.second.get();
+        total_helper_count =
+            static_cast<size_t>(global_helper_count) + static_cast<size_t>(program_info->count_of_helpers);
+        if (total_helper_count < global_helper_count || total_helper_count < program_info->count_of_helpers) {
+            result = EBPF_ARITHMETIC_OVERFLOW;
+            goto Exit;
+        }
+        total_helper_size = total_helper_count * sizeof(ebpf_helper_function_prototype_t);
+        new_helpers = (ebpf_helper_function_prototype_t*)ebpf_allocate(total_helper_size);
+        if (new_helpers == nullptr) {
+            result = EBPF_NO_MEMORY;
+            break;
+        }
+
+        memset(new_helpers, 0, total_helper_size);
+
+#pragma warning(push)
+#pragma warning(disable : 6386) // Buffer overrun while writing to 'new_helpers':  the writable size is
+                                // 'total_helper_size' bytes, but '80' bytes might be written
+        // Copy the global helpers to the new helpers.
+        for (uint32_t i = 0; i < global_helper_count; i++) {
+            new_helpers[i] = global_helpers[i];
+            auto name = _strdup(global_helpers[i].name);
+            if (name == nullptr) {
+                result = EBPF_NO_MEMORY;
+                goto Exit;
+            }
+            new_helpers[i].name = name;
+        }
+#pragma warning(pop)
+
+        if (program_info->count_of_helpers > 0) {
+            memcpy(
+                new_helpers + global_helper_count,
+                program_info->helper_prototype,
+                (program_info->count_of_helpers * sizeof(ebpf_helper_function_prototype_t)));
+            ebpf_free(program_info->helper_prototype);
+        }
+
+        program_info->helper_prototype = new_helpers;
+        program_info->count_of_helpers = (uint32_t)total_helper_count;
+    }
+
+Exit:
+#pragma warning(push)
+#pragma warning(disable : 6385)
+#pragma warning(disable : 6001)
```

Comment:
Recommend adding a // comment saying what the warning is that is being disabled, like you did in lines 502, 554, etc.

Reply:
Done.